File: frontend/src/schema.ts
export class OrganizationSchema {
    id: string = '';
    name: string = '';
    description: string = '';
    createdAt: string = '';
    updatedAt: string = '';
}

export class UserSchema {
    id: string = '';
    email: string = '';
    firstName: string = '';
    lastName: string = '';
    photoURL: string = '';
    permission: string = '';
    organization: string = '';
    organizationPermission: string = '';
    createdAt: string = '';
    updatedAt: string = '';
}

export class LargeModelSchema {
    id: string = '';
    name: string = '';
    description: string = '';
    rank: number = 0;
    provider: string = '';
    organization: string = '';
    active: boolean = false;
    createdAt: string = '';
    updatedAt: string = '';
}

export class ConnectorSchema {
    id: string = '';
    name: string = '';
    description: string = '';
    function: string = '';
    organization: string = '';
    active: boolean = false;
    createdAt: string = '';
    updatedAt: string = '';
}

export class PromptTemplateSchema {
    id: string = '';
    title: string = '';
    description: string = '';
    instructions: string = '';
    organization: string = '';
    template: string = '';
    placeholder: string = '';
    model: string = '';
    connector: string = '';
    active: boolean = false;
    createdAt: string = '';
    updatedAt: string = '';
}

export class ChatSchema {
    id: string = '';
    user: UserSchema = new UserSchema();
    model: LargeModelSchema = new LargeModelSchema();
    template: PromptTemplateSchema = new PromptTemplateSchema();
    history: MessageSchema[] = [];
    organization: string = '';
    createdAt: string = '';
    updatedAt: string = '';
}

export class MessageSchema {
    id?: string = '';
    role: string = '';
    content: string = '';
    createdAt: string = '';
    updatedAt: string = '';
}

export class CompletionsInputSchema {
    query: string = '';
    prompt?: string = '';
    history?: MessageSchema[] = [];
}

File: app/schema.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class OrganizationSchema(BaseModel):
    id: str
    name: str
    description: str
    createdAt: datetime
    updatedAt: datetime

class UserSchema(BaseModel):
    id: str
    email: str
    firstName: str
    lastName: str
    photoURL: str
    permission: str
    organization: str
    organizationPermission: str
    createdAt: datetime
    updatedAt: datetime

class LargeModelSchema(BaseModel):
    id: str
    name: str
    description: str
    rank: int
    provider: str
    organization: str
    active: bool
    createdAt: datetime
    updatedAt: datetime

class ConnectorSchema(BaseModel):
    id: str
    name: str
    description: str
    function: str
    organization: str
    active: bool
    createdAt: datetime
    updatedAt: datetime

class PromptTemplateSchema(BaseModel):
    id: str
    title: str
    description: str
    instructions: str
    organization: str
    template: str
    placeholder: str
    # model: LargeModelSchema
    model: str
    connector: str
    active: bool
    createdAt: datetime
    updatedAt: datetime

class ChatSchema(BaseModel):
    id: str
    user: UserSchema
    model: LargeModelSchema
    template: PromptTemplateSchema
    organization: OrganizationSchema
    history: list[dict]
    organization: str
    createdAt: datetime
    updatedAt: datetime

class MessageSchema(BaseModel):
    id: Optional[str]
    role: str
    content: str
    createdAt: datetime
    updatedAt: datetime

class CompletionsInputSchema(BaseModel):
    """
    This is the schema for the input data to the get_completions endpoint.
    """
    query: str
    prompt: Optional[str]
    history: Optional[list[MessageType]]
